IS0402Test.py

```diff
@@ -816,6 +819,243 @@ def test_30(self):
+                    json_msg = json.loads(curr_msg)
+                    grain_data.extend(json_msg["grain"]["data"])
+
+                found_data_set = False
+                for curr_data in grain_data:
+                    pre_data = json.dumps(curr_data["pre"], sort_keys=True)
+                    post_data = json.dumps(curr_data["post"], sort_keys=True)
+                    sorted_resource_data = json.dumps(resource_data, sort_keys=True)
+                    sorted_old_resource_data = json.dumps(old_resource_data[resource], sort_keys=True)
+
+                    if pre_data == sorted_old_resource_data:
+                        if post_data == sorted_resource_data:
+                            found_data_set = True
+
+                if not found_data_set:
+                    return test.FAIL("Did not found expected data set in websocket UPDATE message for '{}'"
+                                     .format(resource))
+
+            # Verify if corresponding message received via websocket: DELETE
+            reversed_resource_list = copy(resources_to_post)
+            reversed_resource_list.reverse()
+            for resource in reversed_resource_list:
+                valid, r = self.do_request("DELETE", self.reg_url + "resource/{}s/{}".format(resource,
+                                                                                            test_data[resource]["id"]))
+                if not valid or r.status_code != 204:
+                    return test.FAIL("Registration API did not respond as expected: Cannot delete {}: {} {}"
+                                     .format(resource, r.status_code, r.text))
+
+            sleep(1)
+            for resource, resource_data in test_data.items():
+                received_messages = websockets[resource].get_messages()
+
+                # Verify data inside messages
+                grain_data = list()
+
+                for curr_msg in received_messages:
+                    json_msg = json.loads(curr_msg)
+                    grain_data.extend(json_msg["grain"]["data"])
+
+                found_data_set = False
+                for curr_data in grain_data:
+                    pre_data = json.dumps(curr_data["pre"], sort_keys=True)
+                    sorted_resource_data = json.dumps(resource_data, sort_keys=True)
+
+                    if pre_data == sorted_resource_data:
+                        if "post" not in curr_data:
+                            found_data_set = True
+
+                if not found_data_set:
+                    return test.FAIL("Did not found expected data set in websocket DELETE message for '{}'"
+                                     .format(resource))
+
+            # Verify if corresponding message received via Websocket: ADD
+            # Post sample data again
+            for resource in resources_to_post:
+                valid, r = self.do_request("POST", self.reg_url + "resource", data={"type": resource,
```

Comment:
It would be elegant to post the resource data with a bumped version timestamp, rather than the original one.